producer/ue_authentication.go

```diff
@@ -27,6 +27,16 @@ import (
 	"github.com/omec-project/openapi/models"
 )
 
+// Generates a random int between 0 and 255
+func GenerateRandomNumber() (uint8, error) {
+	randBytes := make([]byte, 1)
+	_, err := rand.Read(randBytes)
+	if err != nil {
+		return 0, fmt.Errorf("error while generating random number: %s", err.Error())
+	}
+	return randBytes[0], nil
+}
```

Comment:
Why having a "different" function compared to the one proposed/merged in [NRF #52](https://github.com/omec-project/nrf/pull/52) (also, shown below))? Should not function be pretty much the same as the one in the NRF repo?

```go
// Generates a random int between 0 and 99
func GenerateRandomNumber() (int, error) {
	max := big.NewInt(100)
	randomNumber, err := rand.Int(rand.Reader, max)
	if err != nil {
		return 0, err
	}
	return int(randomNumber.Int64()), nil
}
```

Reply:
I changed this to follow the same approach as in the NRF. I used to use bytes because we're outputting a value between 0-255 instead of an arbitrary value but it doesn't really make a difference. This is more standard with the last commit.